### PR TITLE
RNMT-1020 Added JS script to inject "viewport-fit=cover" tag

### PR DESCRIPTION
**Pull request by pmaroco**

### CODE REVIEW

File: www/statusbar.js [hunk 1]
```diff
@@ -153,6 +156,21 @@ var addStatusBarDataElement = function(){
 }
 
 
+var injectViewportMetaTag = function(){
+
+    if(device.platform.toLowerCase().indexOf(IOS_PLATFORM) == 0 && device.version.split(".")[0].indexOf(IOS_11_VERSION) == 0){
```
Comment (vmfo22) [inline www/statusbar.js]: It's better to validate for >= 11 instead of only for iOS 11, otherwise, with iOS 12 (when released) it should break

**Review by vmfo22 — APPROVED** [review]

**Review by EiyuuZack — APPROVED** [review]

File: www/statusbar.js [hunk 2]
```diff
@@ -153,6 +156,21 @@ var addStatusBarDataElement = function(){
 }
 
 
+var injectViewportMetaTag = function(){
+
+    if(device.platform.toLowerCase().indexOf(IOS_PLATFORM) == 0 && device.version.split(".")[0] >= IOS_11_VERSION){
```
Comment (luissilvaos) [inline www/statusbar.js]: Is this device object provided by the Device plugin?

**Review by luissilvaos — CHANGES_REQUESTED** [review]
Just one question :)

File: www/statusbar.js (hunk 2 above)
Comment (pmaroco) [reply www/statusbar.js]: Yes. https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-device/

**Review by bgracio — APPROVED** [review]

**Review by EiyuuZack — APPROVED** [review]

**Review by luissilvaos — APPROVED** [review]

File: www/statusbar.js [hunk 3]
```diff
@@ -153,16 +152,19 @@ var addStatusBarDataElement = function(){
...
 
 
 var injectViewportMetaTag = function(){
 
-    if(device.platform.toLowerCase().indexOf(IOS_PLATFORM) == 0 && device.version.split(".")[0] >= IOS_11_VERSION){
-        var viewportMetaElem = document.getElementsByTagName("meta").namedItem("viewport");
+    if (/(iPad)|(iPhone)/i.test(navigator.userAgent)) {
+        var version = navigator.appVersion.match(/OS (\d+)_(\d+)_?(\d+)?/);
+
+        if((version && version[1]) >= IOS_11_VERSION) {
```
Comment (bgracio) [inline www/statusbar.js]: Don't you want this: if(version && (version[1] >= IOS_11_VERSION)) ??

**Review by bgracio — CHANGES_REQUESTED** [review]

File: www/statusbar.js (hunk 3 above)
Comment (bgracio) [reply www/statusbar.js]: Also, it seems that IOS_11_VERSION is a string. Comparing strings as they were ints can lead us to future problems.

image

File: www/statusbar.js [hunk 4]
```diff
@@ -161,7 +161,7 @@ var injectViewportMetaTag = function(){
     if (/(iPad)|(iPhone)/i.test(navigator.userAgent)) {
         var version = navigator.appVersion.match(/OS (\d+)_(\d+)_?(\d+)?/);
 
-        if((version && version[1]) >= IOS_11_VERSION) {
+        if(Number(version && version[1]) >= IOS_11_VERSION) {
```
Comment (luissilvaos) [inline www/statusbar.js]: This will "crash" if the version is undefined.

**Review by luissilvaos — CHANGES_REQUESTED** [review]

**Review by bgracio — APPROVED** [review]
✨ ✨ ✨

**Review by luissilvaos — APPROVED** [review]

File: www/statusbar.js [hunk 5]
```diff
@@ -161,7 +161,7 @@ var injectViewportMetaTag = function(){
     if (/(iPad)|(iPhone)/i.test(navigator.userAgent)) {
         var version = navigator.appVersion.match(/OS (\d+)_(\d+)_?(\d+)?/);
 
-        if(version != null && !isNaN(version[1])){
+        if(Array.isArray(version) && version.length > 1 && !isNaN(version[1])){
```
Comment (EiyuuZack) [inline www/statusbar.js]: If `version[1]` is actually a number this gets skipped? 😕

**Review by EiyuuZack — CHANGES_REQUESTED** [review]
See my question.

**Review by EiyuuZack — APPROVED** [review]

**Comment by EiyuuZack** [comment]
#closeallthemergedbranches

**Pull request merged**